doc/specs/msgstake_flow.md

```diff
@@ -0,0 +1,41 @@
+# Flowchart of Output Address Edit
+
+A staked node's output address can be changed only when a transaction message
+`MsgStake` is signed by the owner of the node's current output address.  If a
+transaction is signed by the operator address, or the new output address
+specified in the transaction, it's rejected by the blockchain.  The following
+chart illustrates how a `MsgStake` is handled by the key functions.
+
+```mermaid
+flowchart LR
+  A(baseApp/runTx) --> B(x/auth/ValidateTransaction)
+	B --> C{After the upgrade?}
+	C -- yes --> D{Signed by any of:\nthe new output address\nthe operator address\nthe current output address}
+  D -- yes --> F([ok])
+  D -- no --> G([sdk:4])
+	C -- no --> E{Signed by any of:\nthe new output address\nthe operator address}
+  E -- yes --> F
+  E -- no --> G
+  A --> H(baseApp/runMsg)
```

Comment:
Don't be afraid to split into multiple smaller diagrams. It makes it easier to navigate, share and read.

For example, just scroll (no reason to dive into the details) of https://github.com/pokt-network/pocket/wiki/Utility-Relay-Protocol